test/python/opflow/test_gradients.py

```diff
@@ -1356,6 +1460,132 @@ def test_overlap_qfi_raises_on_unsupported_gate(self):
         with self.assertRaises(NotImplementedError):
             _ = QFI("overlap_diag").convert(StateFn(circuit), [x])
 
+    @data(-Y, Z - 1j * Y)
+    def test_aux_meas_op(self, aux_meas_op):
+        """Test various auxiliary measurement operators for probability gradients with LinComb
+        Gradient
+
+        """
+
+        a = Parameter("a")
+        b = Parameter("b")
+        params = [a, b]
+
+        q = QuantumRegister(1)
+        qc = QuantumCircuit(q)
+        qc.h(q)
+        qc.rz(params[0], q[0])
+        qc.rx(params[1], q[0])
+
+        op = CircuitStateFn(primitive=qc, coeff=1.0)
+
+        shots = 10000
+
+        prob_grad = LinComb().convert(operator=op, params=params, aux_meas_op=aux_meas_op)
+        value_dicts = [{a: [np.pi / 4], b: [0]}, {a: [np.pi / 2], b: [np.pi / 4]}]
+        if aux_meas_op == -Y:
+            correct_values = [
+                [[-0.5, 0.5], [-1 / (np.sqrt(2) * 2), -1 / (np.sqrt(2) * 2)]],
+                [[-1 / (np.sqrt(2) * 2), 1 / (np.sqrt(2) * 2)], [0, 0]],
+            ]
+        else:
+            correct_values = [
+                [[-0.5j, 0.5j], [(1 - 1j) / (np.sqrt(2) * 2), (-1 - 1j) / (np.sqrt(2) * 2)]],
+                [
+                    [-1j / (np.sqrt(2) * 2), 1j / (np.sqrt(2) * 2)],
+                    [1 / (np.sqrt(2) * 2), -1 / (np.sqrt(2) * 2)],
+                ],
+            ]
```

Comment:
Can we pass `correct_values` also in @data? Would save us from using `if`.

Reply:
I don't think that this is necessary.

Reply:
It does not scale nicely if we decided to support more measurement operators.

Reply:
But it is a good pattern to change data, not code, if possible.

Reply:
Ok